src/commands/upload_dart_symbol_map.rs

```diff
@@ -0,0 +1,167 @@
+    }
+}
+
+pub fn make_command(command: Command) -> Command {
+    command
+        .about("Upload a Dart/Flutter symbol map (dartsymbolmap) for deobfuscating Dart exception types.")
+        .after_help(
+            "Examples:\n  sentry-cli dart-symbol-map upload --org my-org --project my-proj path/to/dartsymbolmap.json path/to/debug/file\n\n  The mapping must be a JSON array of strings with an even number of entries (pairs).\n  The debug file must contain exactly one Debug ID.",
+        )
+        .org_arg()
+        .project_arg(false)
+        .arg(
+            Arg::new("mapping")
+                .value_name("MAPPING")
+                .help("Path to the dartsymbolmap JSON file (e.g. dartsymbolmap.json). Must be a JSON array of strings with an even number of entries (pairs).")
+                .required(true),
+        )
+        .arg(
+            Arg::new("debug_file")
+                .value_name("DEBUG_FILE")
+                .help("Path to the corresponding debug file to extract the Debug ID from. The file must contain exactly one Debug ID.")
+                .required(true),
+        )
+}
+
+pub fn execute(matches: &ArgMatches) -> Result<()> {
+    // Parse required positional arguments
+    let mapping_path = matches
+        .get_one::<String>("mapping")
+        .expect("required argument 'mapping' not provided by clap");
+    let debug_file_path = matches
+        .get_one::<String>("debug_file")
+        .expect("required argument 'debug_file' not provided by clap");
+
+    // Extract Debug ID(s) from the provided debug file
+    let dif = DifFile::open_path(debug_file_path, None)?;
+    let mut ids: Vec<_> = dif.ids().into_iter().filter(|id| !id.is_nil()).collect();
+
+    // Ensure a single, unambiguous Debug ID
+    ids.sort();
+    ids.dedup();
+    match ids.len() {
+        0 => bail!(
+            "No debug identifier found in the provided debug file ({}). Ensure the file contains an embedded Debug ID.",
+            debug_file_path
+        ),
+        1 => {
+            let debug_id = ids.remove(0);
+
+            // Validate the dartsymbolmap JSON: must be a JSON array of strings with even length
+            let mapping_file_bytes = std::fs::read(mapping_path)
+                .with_context(|| format!("Failed to read mapping file at {mapping_path}"))?;
+            let mapping_entries: Vec<String> = serde_json::from_slice(&mapping_file_bytes)
+                .context("Invalid dartsymbolmap: expected a JSON array of strings")?;
+
+            if mapping_entries.len() % 2 != 0 {
+                bail!(
+                    "Invalid dartsymbolmap: expected an even number of entries (pairs), got {}",
+                    mapping_entries.len()
+                );
+            }
+
+            // Prepare upload object
+            let file_name = Path::new(mapping_path)
+                .file_name()
+                .and_then(OsStr::to_str)
+                .unwrap_or(mapping_path)
+                .to_owned();
```

Comment:
You can avoid a copy here

```suggestion
            let file_name = Path::new(mapping_path)
                .file_name()
                .and_then(OsStr::to_str)
                .unwrap_or(mapping_path);
```